src/main/java/org/stellar/sdk/SorobanServer.java

```diff
@@ -560,6 +544,52 @@ public SendTransactionResponse sendTransaction(Transaction transaction) {
         "sendTransaction", params, new TypeToken<SorobanRpcResponse<SendTransactionResponse>>() {});
   }
 
+  public GetSacBalanceResponse getSacBalance(String contractId, Asset asset, Network network)
+      throws IOException {
+
+    if (!StrKey.isValidContract(contractId)) {
+      throw new IllegalArgumentException("expected contract ID, got " + contractId);
+    }
+
+    LedgerKey ledgerKey =
+        LedgerKey.builder()
+            .discriminant(LedgerEntryType.CONTRACT_DATA)
+            .contractData(
+                LedgerKey.LedgerKeyContractData.builder()
+                    .contract(Scv.toAddress(asset.getContractId(network)).getAddress())
+                    .key(
+                        Scv.toVec(
+                            Arrays.asList(Scv.toSymbol("Balance"), Scv.toAddress(contractId))))
+                    .durability(ContractDataDurability.PERSISTENT)
+                    .build())
+            .build();
+
+    GetLedgerEntriesResponse response = this.getLedgerEntries(Collections.singleton(ledgerKey));
+
+    List<GetLedgerEntriesResponse.LedgerEntryResult> entries = response.getEntries();
+    if (entries == null || entries.isEmpty()) {
+      return null;
+    }
+
+    GetLedgerEntriesResponse.LedgerEntryResult entry = entries.get(0);
+    LedgerEntry.LedgerEntryData ledgerEntryData =
+        LedgerEntry.LedgerEntryData.fromXdrBase64(entry.getXdr());
```

Comment:
Please use `Utils.parseXdr`, and do not throw `IOException` in the function signature.